nat-lab/tests/test_upnp_connection.py

```diff
@@ -17,6 +17,22 @@
 from utils.router import new_router, IPStack
 
 
+async def execute_upnpc_with_retry(connection, timeout=10.0):
+    start_time = asyncio.get_event_loop().time()
+    while True:
+        cmd = await connection.create_process(["upnpc", "-i", "-l"]).execute()
```

Comment:
This would raise an exception, breaking your loop and test in general. Instead of checking for return code, you have to catch `ProcessExecError` and then only try retry logic.

Reply:
Wrapped